### PR TITLE
executor: support abort the cluster log retriever

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

This PR adds support abort retrieve log from the remote component.

### What is changed and how it works?

Binding cancel function with context and cancel the subsequent progress while the retriever closed.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: executor/cluster_reader.go [hunk 1]
```diff
@@ -47,10 +47,15 @@ import (
 	"google.golang.org/grpc/credentials"
 )
 
-const clusterLogBatchSize = 1024
+const clusterLogBatchSize = 256
+
+type dummyClose struct{}
```
Comment (winkyao) [inline executor/cluster_reader.go]: dummyCloser?

File: executor/cluster_reader.go [hunk 2]
```diff
@@ -370,6 +382,7 @@ type clusterLogRetriever struct {
 	retrieving bool
 	heap       *logResponseHeap
 	extractor  *plannercore.ClusterLogTableExtractor
+	cancel     func()
```
Comment (Deardrops) [inline executor/cluster_reader.go]:
```suggestion
	cancel     context.CancelFunc
```

Comment (Deardrops) [inline executor/cluster_reader.go]:
```suggestion
	cancel     context.CancelFunc
```

**Pull request merged**